cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml

```diff
@@ -0,0 +1,272 @@
+    nextStep: Start_Build
+  - name: Go_Build_Enviroment
+    action: 'aws:executeAwsApi'
+    inputs:
+      Service: codebuild
+      Api: CreateProject
+      artifacts:
+        type: NO_ARTIFACTS
+      description: This environment builds a 3P resource
+      environment:
+        computeType: BUILD_GENERAL1_SMALL
+        image: 'aws/codebuild/amazonlinux2-x86_64-standard:4.0'
+        type: LINUX_CONTAINER
+        privilegedMode: true
+      name: '{{ProjectName}}'
+      serviceRole: '{{ BuilderRole }}'
+      timeoutInMinutes: 480
+      source:
+        buildspec: |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.9
+                golang: 1.18
+              commands:
+                - pip install pre-commit
+                - yum update -y
+                - yum install docker -y
+            build:
+              commands:
+                - aws --version
+                - wget "https://github.com/aws/aws-sam-cli/releases/latest/download/aws-sam-cli-linux-x86_64.zip"
+                - unzip aws-sam-cli-linux-x86_64.zip -d sam-installation
+                - ./sam-installation/install
+                - pip3 install cloudformation-cli cloudformation-cli-go-plugin
+                - curl  https://load-gen-builds-peerislands.s3.amazonaws.com/mongodb-enterprise-6.0.repo > /etc/yum.repos.d/mongodb-enterprise-6.0.repo
+                - sudo yum install -y mongodb-atlas-cli
+                - export ATLAS_ORG_ID={{OrgID}}
+                - export ATLAS_PUBLIC_KEY={{PubKey}}
+                - export ATLAS_PRIVATE_KEY={{PvtKey}}
+                - export MCLI_PUBLIC_API_KEY=$ATLAS_PUBLIC_KEY
+                - export MCLI_PRIVATE_API_KEY=$ATLAS_PRIVATE_KEY
+                - export MCLI_ORG_ID=$ATLAS_ORG_ID
+                - export MCLI_OUTPUT=json
+                - export AWS_DEFAULT_REGION={{Get_Region.RegionName}}
+                - 'git clone --branch {{ BranchName }}  {{ Repository }} provider'
+                - cd provider
+                - cd cfn-resources
+                - echo "Starting cfn-publish.sh"
+                - ./cfn-publish.sh {{ ResourceName }} {{OtherParams}}
+        type: NO_SOURCE
+    description: Creates an environment that deploys a 3rd party resource
+    nextStep: Start_Build
+  - name: Start_Build
+    action: 'aws:executeAwsApi'
+    inputs:
+      Service: codebuild
+      Api: start-build
+      projectName: '{{ProjectName}}'
+    outputs:
+      - Selector: build.id
+        Type: String
+        Name: id
+  - name: Await_Build_Status
+    action: 'aws:waitForAwsResourceProperty'
+    inputs:
+      Service: codebuild
+      Api: batch-get-builds
+      ids:
+        - '{{ Start_Build.id }}'
+      PropertySelector: '$.builds[0].buildStatus'
+      DesiredValues:
+        - SUCCEEDED
+        - FAILED
+        - FAULT
+        - STOPPED
+        - TIMED_OUT
+  - name: Check_Build_Status
```

Comment:
Main change is in this step. Have committed current version in the branch for reference.